main.rb

```diff
@@ -0,0 +1,175 @@
+require './book'
+require './student'
+require './teacher'
+require './rental'
+class App
+  def initialize
+    @list_books = []
+    @list_people = []
+    @list_rentals = []
+  end
+
+  # Books Block
+  def insert_book
+    print 'Insert Title: '
+    title = gets.chomp
+    print 'Insert Author: '
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @list_books.push(new_book)
+    puts 'Book created Succesfully'
+    show_menu
+  end
+
+  def show_books
+    if @list_books.any? == false
+      puts "There's no books registered, Press 1 to return"
+      selected = gets.chomp
+      show_menu if selected == '1'
+    else
+      @list_books.each { |b| puts "Title: #{b.title} Author: #{b.author}" }
+      puts 'Add another Book ? [1:yes no:2]'
+      option = gets.chomp
+      if option == '1'
+        insert_book
+      else
+        show_menu
+      end
+    end
+  end
+
+  #============================
+  # People Block
+  def insert_student
+    print 'Insert Name:'
+    name = gets.chomp
+    print 'Insert Age:'
+    age = gets.chomp
+    print 'Has permission? [y, n] '
+    permission = gets.chomp
+    permission = permission == 'y'
+    @list_people.push(Student.new(age, name, permission))
+    puts 'Student created Succesfully'
+    show_menu
+  end
+
+  def insert_teacher
+    print 'Insert Name: '
+    name = gets.chomp
+    print 'Insert Age: '
+    age = gets.chomp
+    print 'Insert Specialization: '
+    spec = gets.chomp
+    @list_people.push(Teacher.new(age, name, spec))
+    puts 'Teacher created Succesfully'
+    show_menu
+  end
```

Comment:
- [ ] great job creating the UI 👍 However when creating a student or teacher the user received below error, please fix this before requesting another code review.

![afbeelding](https://user-images.githubusercontent.com/39240820/146588969-81c9539b-f023-455b-b0a7-c674144f82bc.png)

Reply:
thanks, resolve it!